### ISSUE TITLE
Project Generator/Workflow Guide

### ISSUE THREAD

**Issue by pixelknitter** [codepath/nodejs_guides#18]
There are a lot of options out there. One thing I found challenging when first starting out was what a production level workflow looked like, I started out exploring the yeoman workflow:

http://yeoman.io/

**What success looks like: Succinctness and clarity. Less is more.**
Reference Wikis: [Authentication & Authorization](https://github.com/codepath/nodejs_guides/wiki/Authorization-&-Authentication), [Setup](https://github.com/codepath/nodejs_guides/wiki/Setup)

Open separate issues for required links to community sources if you're having trouble finding any.

_Please assign to yourself before working on this._
/cc @NinjaSudo @philster @azoff @DiyahM @CrabDude

**Comment by CrabDude** [comment]
I really like generators in theory, but there don't tend to be as many as I wish existed, and many of them are borderline useless. If you have more experience, I would be happy to have a guide for them, though it's much lower priority than the other guides because I won't be using generators in any of the projects/classes.

We want students to learn how to build projects from scratch so they understand how the pieces fit together. That said, week 5 is focused on production / deployment, and this could fit nicely in either the lecture or the exercise. There's not much time in either, but generators can be pretty convenient in an enterprise environment.

**Comment by pixelknitter** [comment]
Agreed building from scratch is important. I'm no expert with generators, but could be something to research and put together a well organized guide for the class follow along with the lectures / exercises. I've definitely felt many opinions on these tools and project generators to spin up a new project.

[Gulp](http://gulpjs.com/) is what I've heard to be the new hotness for production / deployment automation.

**Comment by CrabDude** [comment]
`gulp` would fall into tooling, and will definitely be covered week 5.
